com.unity.netcode.gameobjects/Components/NetworkTransform.cs

```diff
@@ -532,7 +537,12 @@ private void ApplyInterpolatedNetworkStateToTransform(NetworkTransformState netw
             // again, we should be using quats here
             if (SyncRotAngleX || SyncRotAngleY || SyncRotAngleZ)
             {
-                var eulerAngles = m_RotationInterpolator.GetInterpolatedValue().eulerAngles;
+                var eulerAngles = new Vector3();
```

Comment:
I guess this touches less code, but it does seem a little odd to create this and then never use it in lines 546-560. 

What if instead of all the 
`Interpolate ? networkState.Rotation.x : eulerAngles.x;`

You just do ones
`rotationTmp = Interpolate ? networkState.Rotation : eulerAngles;`

And then these turn into
```
interpolatedRotAngles.x = networkState.IsTeleportingNextFrame || rotationTmp.x;
```

Reply:
Hi Matt, if I read you correctly, you took:

`interpolatedRotAngles.x = networkState.IsTeleportingNextFrame || !Interpolate ? networkState.Rotation.x : eulerAngles.x;`

and suggest splitting it in two parts:

`interpolatedRotAngles.x = networkState.IsTeleportingNextFrame || ...`

and pre-computing the part after the logical or. This doesn't quite work. Operation precedence makes it prioritize this way:

`interpolatedRotAngles.x = (networkState.IsTeleportingNextFrame || !Interpolate) ? networkState.Rotation.x : eulerAngles.x;

And the resulting code you suggested:

`interpolatedRotAngles.x = networkState.IsTeleportingNextFrame || rotationTmp.x;`

assigns to a `float` the logical or between a `bool` and a `float`. Ok if I dismiss this suggestion?